dev/bots/analyze_framework_code.dart

```diff
@@ -0,0 +1,421 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:analyzer/dart/analysis/analysis_context.dart';
+import 'package:analyzer/dart/analysis/analysis_context_collection.dart';
+import 'package:analyzer/dart/analysis/results.dart';
+import 'package:analyzer/dart/analysis/session.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:analyzer/source/line_info.dart';
+import 'package:collection/collection.dart';
+import 'package:path/path.dart' as path;
+
+import 'utils.dart';
+
+Future<void> runVerifiersInResolvedDirectory(String workingDirectory, List<ResolvedUnitVerifier> verifiers) async {
+  final String flutterLibPath = path.canonicalize('$workingDirectory/packages/flutter/lib');
+  final AnalysisContextCollection collection = AnalysisContextCollection(
+    includedPaths: <String>[flutterLibPath],
+    excludedPaths: <String>[path.canonicalize('$flutterLibPath/fix_data')],
+  );
+
+  final List<String> analyzerErrors = <String>[];
+  for (final AnalysisContext context in collection.contexts) {
+    final Iterable<String> analyzedFilePaths = context.contextRoot.analyzedFiles();
+    final AnalysisSession session = context.currentSession;
+
+    for (final String filePath in analyzedFilePaths) {
+      final SomeResolvedUnitResult unit = await session.getResolvedUnit(filePath);
+      if (unit is ResolvedUnitResult) {
+        for (final ResolvedUnitVerifier verifier in verifiers) {
+          verifier.analyzeResolvedUnitResult(unit, workingDirectory);
+        }
+      } else {
+        analyzerErrors.add('Analyzer error: file $unit could not be resolved.');
+      }
+    }
+  }
+
+  if (analyzerErrors.isNotEmpty) {
+    foundError(analyzerErrors);
+  }
+  for (final ResolvedUnitVerifier verifier in verifiers) {
+    verifier.reportError();
+  }
+}
+
+abstract class ResolvedUnitVerifier {
+  void reportError();
+  void analyzeResolvedUnitResult(ResolvedUnitResult unit, String workingDirectory);
```

Comment:
Nit: should we stick with the "verify" terminology can call this "verifyResolvedUnitResult"?

Reply:
renamed the class to `AnalyzeRule`